core/src/main/python/wlsdeploy/tool/util/mbean_utils.py

```diff
@@ -0,0 +1,767 @@
+                        found = False
+        _logger.exiting(class_name=self.__class__.__name__, method_name=_method_name, result=Boolean(found))
+        return found
+
+    def is_lsa_in_attributes(self, lsa_attribute, attribute_list):
+        """
+        Look for the offline WLST LSA attribute name in one of the MBean helper lists. The names
+        can differ between the LSA attribute and the MBean attribute lists. Attempt to match the LSA
+        attribute using different representations of the name.
+        :param lsa_attribute: attribute from the offline WLST LSA list
+        :param attribute_list: list of attributes from one of the MBean helper maps
+        :return: True if the LSA attribute is found in the MBean list
+        """
+        _method_name = 'is_lsa_in_attributes'
+        _logger.entering(lsa_attribute, class_name=self.__class__.__name__, method_name=_method_name)
+        found = True
+        if lsa_attribute not in attribute_list and not self.__is_found_with_lower_case(lsa_attribute, attribute_list):
+            if not lsa_attribute.endswith('y') or not \
+                   self.__is_found_with_lower_case(lsa_attribute[:len(lsa_attribute) - 1] + 'ies', attribute_list):
+                if not self.__is_found_with_lower_case(lsa_attribute + 'es', attribute_list):
+                    if not self.__is_found_with_lower_case(lsa_attribute + 's', attribute_list):
+                        found = False
+        _logger.exiting(class_name=self.__class__.__name__, method_name=_method_name, result=Boolean(found))
+        return found
+
+    def __is_found_with_lower_case(self, attribute, mbean_list):
+        _method_name = '_is_found_with_lower_case'
+        _logger.entering(attribute, class_name=self.__class__.__name__, method_name=_method_name)
+
+        found = False
+        try:
+            found = len([key for key in mbean_list if key.lower() == attribute.lower()]) > 0
+        except (ValueError, KeyError):
+            pass
+
+        _logger.exiting(class_name=self.__class__.__name__, method_name=_method_name, result=Boolean(found))
+        return found
+
+
+class MBeanAttributes(object):
+    """
+    The MBeanAttributes class is a wrapper class encapsulating MBean information for the different types of
+    MBean attribute collections. An instance of this class contains all attributes for an MBean at a specific
+    location. The methods in this class operates on the attributes the class encapsulates to provide attribute
+    specific information.
+    """
+
+    __interface_matcher = re.compile('Bean$')
+
+    def __init__(self, model_context, exception_type, location):
+        self.__model_context = model_context
+        self.__exception_type = exception_type
+        self.__location = location
+        self.__aliases = Aliases(self.__model_context, wlst_mode=self.__model_context.get_target_wlst_mode())
```

Comment:
I think we should not construct the Aliases object, but pass it in.  It takes some effort to re-build it per WLS version.  Maybe just pass AliasHelper around, it should have all the methods, and the exception type is baked in.

Reply:
I would be nice to pull that exception type OUT of the model_context - would be useful for any place that would like to generate and throw a tool level  exception. It's being passed around to those classes that do not generate their own exception types.

Speaking of which, I considered a separate type for this class - but the class is just so small and single use, I decided not to. Opinions?